### ISSUE TITLE
Use entity "ratio" for contrast ratio "3:1" statements

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#1117]
[From Wolfgang Berndorfer](https://lists.w3.org/Archives/Public/public-agwg-comments/2020Apr/0000.html):

> Please use the correct character code for ratio e.g. 3&ratio;1 and not 3:1.
> WCAG 2.1 and the working draft for WCAG 2.2 use : but it should be ∶ (`&ratio;` or `&#8758;`).
> Visually both look nearly the same, but good screen readers announce them differently.

**Comment by guyhickling** [comment]
Some screen readers do not announce the colon at all, at least in their default configuration. They just announce "three one". So it is good accessibility practice to write all ratios as "3 to 1" rather than use the colon symbol. That guarantees correct rendition. I am fairly sure I have even seen that mentioned somewhere in W3C documentation, but on searching just now I can't find it. Be that as it may, if we encourage people to use the word "to" in ratios, that ensures correct announcement by all readers, and would also solve the character problem Wolfgang has raised. 

I think W3C should certainly do this, as standard, in all documentation as we use ratios so often. Could that be put to the relevant groups?

I didn't even know there was a difference in character codes for colons, but I don't think there is anything to be done about them. I am looking at two different PC keyboards right now, from Logitech and Dell, and both show only one colon key. So that's what people will use, and I don't think you would ever have success in asking people to type an ampersand &code when the key is available right in front of them. Another reason for using the word "to" in ratios, I think. 

Deque wrote a [good article on screen readers and symbols](https://www.deque.com/blog/dont-screen-readers-read-whats-screen-part-1-punctuation-typographic-symbols/) a few years ago, and I think we should do as that article says and put pressure on the screen reader makers to announce the important symbols in the default configuration, and allow users to stop that if they prefer not to have it. They should cooperate and all announce things like this the same way. As the article points out "Screen reader flaws breed apathy among web developers and content writers. Such apathy is counter-productive to the goal of creating an accessible web. Too many people will just assume that screen readers are always a little bit broken....." (and people do!)

**Comment by detlevhfischer** [comment]
So the character inserted by the default colon key is the wrong one (at least not read by NVDA). And default Narrator does not seem read ":" whatever character variant is used.
I'd like to point out that while I can see the benefits to replacing the : character with 'to' as suggested by @guyhickling  I can also see a disadvantage for sighted users. Perceptual groupings like 4.5:1 stick out visually - they immediately convey (in a WCAG evaluation context) that we are dealing with a contrast ratio. And I dare wonder whether for screen reader users it is really a significant difference hearing "contrast is four point five to one" instead of "contrast is four point five [pause] one" especially if the latter is what users are already accustomed to due to the default use of the silent colon (I realise that context is key here, and the advantage more pronounced for such values appearing without context).

**Comment by patrickhlauke** [comment]
agree with detlev. i'd also point out that the vast majority of people who then write ratios in audit reports, emails, etc will still end up using the regular ":" when they write stuff...so even IF this is done in the official documentation, it's not going to magically fix it across the internets

**Comment by detlevhfischer** [comment]
@patrickhlauke just updated comment above yours so you may want to check if you still agree... :)

**Comment by patrickhlauke** [comment]
my point still stands that even if the spec does it, nobody else will in their writing etc.

**Comment by guyhickling** [comment]
Thinking about this a bit more, and leaving aside the matter of W3C's own documents which can be amended if desired, so just considering the world's websites, it seems to me we need to be pragmatic about this, rather than try to insist on absolute logical  correctness. Screen readers announce the &ratio character in a very unnatural way, not the way the ordinary people talk about ratios in real  life. Whether we are mathematicians or completely ignorant of the subject, we don't say "4.5 ratio 1", we say "4.5 to 1". Everyone hears the local horse racing results at one time or another and are quite used to stuff like "thirteen to one".

Add that to the fact that the ratio character does not exist on most computer keyboards, so no one is going to start using &ratio unless we embark on a several years-long campaign to educate the world into doing so. Ordinary content writers aren't going to bother with that. They don't understand the difference in the two characters. (I suspect you'll find W3C's personnel might think the same and decide not to do it - it will be a huge job to go through all W3C's pages and change every colour contrast ratio! It isn't just the WCAG itself that has this issue.)

And there are problems in software too. Some programs don't let you enter that character. For the ones that do, I just tried to find out how to do it in MS Word. I tried to follow several instructions on how to use Word's Equations menu, found by Googling; now I may be dumb, but I failed to understand any of them! Of course you can do Alt + Numlock + the number code, but that's asking people to learn and remember the code and the process - and all apparently to no purpose as it appears (to the ordinary person) to produce the same result as the colon key so why not just use that key!

Whereas just telling web content writers that screen readers don't announce colons so can they use "to" instead for ratios, seems to me much more likely to succeed. It's an easy concept for people to grasp, without the need to understand and remember character code numbers and ampersand codes. I ask that of IT teams and they do it!

**Comment by bruce-usab** [comment]
IMHO a change should be informed by polling users of screen reading software.  There are so many other situations where using the quote correct unquote character just creates problems.  Humans are well experienced with subconsciously filling in the gaps, and premature migration to the quote correct unquote character may introduce a barrier to our core audience.  In this case, the screen reading software will still the vocalization wrong.  A human reading aloud will say three to one ratio.  The screen reader, with the exact Unicode, will say three ratio one.  What  A human hearing quote three colon one unquote has no problem.

I speak from experience, as our site is struggling with what to do for inch marks.  Ping to my colleague @kengdoj who also is a now a firm vote for U.S.A. moving to the metric system.

**Comment by alastc** [comment]
>  A human hearing quote three colon one unquote has no problem.

From the table above "none" means no output. What you get isn't "three colon one", it is "three one", with essentially a space as a separator. 

I think "3 ratio 1" is logically an improvement, even if the wording is a bit odd. I'll ping a couple of people who use screenreaders properly to see what they think.

I think consistency is key whichever way we go, so thanks to @fstrr for tackling all of them in one go.

**Comment by LJWatson** [comment]
As a screen reader user I do not think that switching to use the ratio character is worth it, mostly for the reasons that @patrickhlauke and @guyhickling have suggested.

Which punctuation marks screen readers do or do not speak is not a terribly useful metric. There are too many variables that change the outcome, including the screen reader, the TTS, the character set, and the punctuation settings the user has chosen.

Also, screen reader users are quite accustomed to inspecting things more closely when they do not speak as expected, then learning from that experience. We do this every time we want to know how to spell an unfamiliar word for example. Hearing "3 1" in this context *might* be a little confusing at first, but a quick exploration will quickly reveal why.

As @patrickhlauke notes, whatever happens in the spec, it's unlikely to gain wide adoption anywhere else, and so screen reader users will end up with the cognitive burden of having the same thing announced differently in different places.

Lastly, the word "ratio" hasthree sylables, where "colon" has two. Not much in the scheme of things, but it's surprising how quickly such things can add up to an excessively verbose experience.

**Comment by alastc** [comment]
Thanks Léonie, I've also had similar comments from another screen reader user. I think we'll leave it as it is then, sorry @fstrr for asking you to tackle this :-( 

@JaninaSajka also mentioned that there is a "[Pronunciation Gap Analysis and Use Cases](https://www.w3.org/TR/pronunciation-gap-analysis-and-use-cases/)" doc from the W3C, and this might be a good use-case to mention. I'll point the original poster to that, and suggest they raise an issue there.

**Comment by alastc** [comment]
Closing, I wrote back to the original poster: https://lists.w3.org/Archives/Public/public-agwg-comments/2022Mar/0000.html

**Comment by WBerndorfer** [comment]
OK, there are some valid arguments against &ratio:
-	Lot of work to change W3C documents and others.
-	More effort for editors since there is not the one keyboard button for input the character.
-	SR users are familiar with (bad) design and usability issues with symbols.

One argument is ignorable for me: Counting syllables. This depends on the language. Think global 😉
I’ll continue to use &ratio experimentally.
